test/wiki-report.py

```diff
@@ -0,0 +1,120 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; If not, see <http://www.gnu.org/licenses/>.
+
+import json
+import logging
+import sys
+from operator import attrgetter
+
+import mwclient.errors  # type: ignore[import-untyped]
+from wikitcms.wiki import ResTuple, Wiki  # type: ignore[import-untyped]
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+class LoginError(Exception):
+    """Raised when cannot log in to wiki to submit results."""
+    pass
+
+
+class WikiReport:
+    def __init__(self, report_path):
+        self.report_path = report_path
+        self.report = self.parse_json_report()
+        self.compose_id = self.report["metadata"]["compose"]
+        self.compose = self.compose_id.split("-")[-1]
+        self.wiki_hostname = "stg.fedoraproject.org"
+
+    def get_passed_testcases(self):
+        passed_testcases = set()
+
+        for testcase in self.report["tests"]:
+            # Skip testcases that don't have an openqa test associated with them
+            if not testcase["openqa_test"]:
+                continue
+
+            if testcase["status"] == "pass":
+                passed_testcases.add(
+                    ResTuple(
+                        bot=True,
+                        cid=self.compose_id,
+                        compose=self.compose,
+                        dist="Fedora",
+                        env=testcase["env"],
+                        milestone="Rawhide",
+                        release="42",
+                        user="anaconda-bot",
+                        section=testcase["wikictms_section"],
+                        status="pass",
+                        testcase=testcase["openqa_test"],
+                        testname=testcase["test_name"],
+                        testtype="Installation",
+                    )
+                )
+
+        return sorted(passed_testcases, key=attrgetter('testcase'))
+
+    def parse_json_report(self):
+        with open(self.report_path, "r") as f:
+            return json.load(f)
+
+    def login(self, wiki):
+        if not wiki.logged_in:
+            # This seems to occasionally throw bogus WrongPass errors
+            try:
+                wiki.login()
+            except mwclient.errors.LoginError:
+                wiki.login()
+
+        if not wiki.logged_in:
+            logger.error("could not log in to wiki")
+            raise LoginError
+
+    def upload_report(self, wiki, passed_testcases):
+        # Submit the results
+        (insuffs, dupes) = wiki.report_validation_results(passed_testcases)
+
+        for dupe in dupes:
+            tmpl = "already reported result for test %s, env %s! Will not report dupe."
+            logger.info(tmpl, dupe.testcase, dupe.env)
```

Comment:
I don't think that `tmpl` variable is necessary, just write the message on two lines or create a new line before.

Just a suggestion, I don't insist on fixing this.

Reply:
it would be long line